conformance_tests/test_tiovx/test_ext_supplementary_data.c

```diff
@@ -0,0 +1,1438 @@
+    ERROR_CHECK_VX_SUCCESS(vxSetSupplementaryUserDataObject((vx_reference)(matrix2), exemplar2), NULL);
+
+    vx_user_data_object supp1 = vxGetSupplementaryUserDataObject((vx_reference)(matrix1), NULL, &status);
+    ERROR_CHECK_VX_SUCCESS(status , NULL);
+    vx_user_data_object supp2 = vxGetSupplementaryUserDataObject((vx_reference)(matrix2), NULL, &status);
+    ERROR_CHECK_VX_SUCCESS(status , NULL);
+    user_data_t test1 = {0};
+    user_data_t test2 = {0};
+    /* Now try a swap */
+    ERROR_CHECK_VX_SUCCESS(vxuSwap(context, (vx_reference)(matrix1), (vx_reference)(matrix2)), NULL);
+    /* Now if we look at the supplementary data it should be the other way around */
+    ERROR_CHECK_VX_SUCCESS(vxCopyUserDataObject(supp1, 0, sizeof(user_data_t), &test1, VX_READ_ONLY, VX_MEMORY_TYPE_HOST), NULL);
+    ERROR_CHECK_VX_SUCCESS(vxCopyUserDataObject(supp2, 0, sizeof(user_data_t), &test2, VX_READ_ONLY, VX_MEMORY_TYPE_HOST), NULL);
+    status = VX_SUCCESS;
+    for (int i = 0; i < 3; ++i)
+    {
+        if (test1.numbers[i] != user_data2.numbers[i] || test2.numbers[i] != user_data1.numbers[i])
+        {
+            status = VX_FAILURE;
+            break;
+        }
+    }
+    ERROR_CHECK_VX_SUCCESS(status, "Supplementary data swapped when objects swapped");
+    /* Now try a Copy */
+    ERROR_CHECK_VX_SUCCESS(vxuCopy(context, (vx_reference)(matrix1), (vx_reference)(matrix2)), NULL);
+    /* And now supp2 should have the same data as supp1, i.e. user_data_2*/
+    ERROR_CHECK_VX_SUCCESS(vxCopyUserDataObject(supp2, 0, sizeof(user_data_t), &test2, VX_READ_ONLY, VX_MEMORY_TYPE_HOST), NULL);
+    status = VX_SUCCESS;
+    for (int i = 0; i < 3; ++i)
+    {
+        if (test2.numbers[i] != user_data2.numbers[i])
+        {
+            status = VX_FAILURE;
+            break;
+        }
+    }
+    ERROR_CHECK_VX_SUCCESS(status, "Supplementary data copied when objects copied");
+    
+    /* Now test copy & swap in a graph */
+    ERROR_CHECK_VX_SUCCESS(vxCopyUserDataObject(supp1, 0, sizeof(user_data_t), &user_data1, VX_WRITE_ONLY, VX_MEMORY_TYPE_HOST), NULL);
+
+    vx_graph graph = vxCreateGraph(context);
+    vx_node node = vxSwapNode(graph, (vx_reference)(matrix1), (vx_reference)(matrix2));
+    ERROR_CHECK_VX_SUCCESS(vxProcessGraph(graph), NULL);
+    ERROR_CHECK_VX_SUCCESS(vxCopyUserDataObject(supp1, 0, sizeof(user_data_t), &test1, VX_READ_ONLY, VX_MEMORY_TYPE_HOST), NULL);
+    ERROR_CHECK_VX_SUCCESS(vxCopyUserDataObject(supp2, 0, sizeof(user_data_t), &test2, VX_READ_ONLY, VX_MEMORY_TYPE_HOST), NULL);
+    status = VX_SUCCESS;
+    for (int i = 0; i < 3; ++i)
+    {
+        if (test1.numbers[i] != user_data2.numbers[i] || test2.numbers[i] != user_data1.numbers[i])
+        {
+            status = VX_FAILURE;
+            break;
+        }
+    }
+    ERROR_CHECK_VX_SUCCESS(status, "Supplementary data swapped when objects swapped in a graph");
+    vxReleaseNode(&node);
```

Comment:
Please add "VX_CALL" around these release calls.  There are several instances of this missing

Reply:
done